### ISSUE TITLE
UI Interaction Strange Behavior

### ISSUE THREAD

**Issue by riker96** [TomorrowTodayLabs/NewtonVR#45]
I have implemented NewtonVR's  NVR Hand script in order to leverage the UI pointer functionality.  In my scene, there is a primary canvas and a secondary canvas that becomes activated in place of the original canvas.  So, two canvases, one initially active, and the second becoming active later.  When the second canvas becomes active, the first becomes deactivated.

Both are world-space canvases.  The first is free standing in the world and pointing at it with either controller works perfectly (thank you).  

The second canvas is a child of the left hand controller, so it is always attached to the left hand.  The problem is that I can not point at this second, hand-attached canvas with the other free hand.  Instead, for some reason, I must 'point' at the UI elements on this second canvas by gazing at them with the VIVE headset.  If I gaze at the canvas, the cursor appears and I can then use the trigger to select whichever button/slider I am gazing at.

I can not figure out why this is happening.  

Can you please let me know if this is a feature or a bug?  I'd like to simply treat this canvas as any other, and use the free controller to interact.

I hope this is clear and I look forward to your reply.

**Comment by riker96** [comment]
UPDATE: The described behavior seems to be caused by introducing a canvas after startup.  If I enable the second canvas from the outset (i.e. it is active and seen by the event system during Start()), then it can be interacted with normally.  So, I've just kept it active and simply switch it's position in and out of view to mimic enable/disable. 

So, the gaze pointer behavior is default for canvases added mid-game, it seems.  Thoughts?

**Comment by adrianrudnik** [comment]
Yes, seems like the foreach GameObject.FindObjectsOfType<Canvas> in Start() of NVRCanvasInput does assign the world camera to each canvas once.

There are cases where the HUD is spawned from prefabs, and those get the wrong camera (hud eyes), so you end up with a misconfigured hand, which looks and acts like a gaze pointer.

Not sure how to approach it, just dug into the topic for 15 minutes and the ControllerCamera is not really accessible. Event systems and broadcasts aside, maybe create a dynamic canvas script that you could attach to your canvas. It should look up the ControllerCamera for the UI and assign it as worldCamera on the canvas.

**Comment by munkeyxis** [comment]
I just updated to Unity 5.6.0f3 and the latest NewtonVR. It seems the coroutine that spawns the Controller UI Camera isn't running beyond the yeild return null and hence Canvases that are enabled upon scene start do not get the Controller UI Camera assigned as their Event Camera.
